### PR TITLE
Require modules from application, not from impress

### PR DESCRIPTION
**Pull request by tshemsedinov**
Closes: https://github.com/metarhia/impress/issues/1707

- [x] tests and linter show no problems (`npm t`)
- [ ] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [ ] description of changes is added in CHANGELOG.md
- [ ] update .d.ts typings

### CODE REVIEW

File: lib/deps.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 'use strict';
 
+const CWD = process.cwd();
+
 const wt = require('node:worker_threads');
+const { createRequire } = require('node:module');
 const metautil = require('metautil');
+const appRequire = createRequire(`file://${CWD}/server.js`);
```
Comment (lundibundi) [inline lib/deps.js:8]: What if the user starts app from some nested folder? Do we perhaps want to try "search up the tree" to find package.json?

File: lib/deps.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
 'use strict';
 
+const CWD = process.cwd();
```
Comment (lundibundi) [inline lib/deps.js:3]: Do we still not support TS application? as this will probably not work with `dist`.

File: lib/deps.js (hunk 1 above)
Comment (tshemsedinov) [reply lib/deps.js:8]: This solution expected  all metarhia projects to be of the same structure and they should have server.js in app root folder and I am ok with this

File: lib/deps.js (hunk 2 above)
Comment (tshemsedinov) [reply lib/deps.js:3]: We don't support TS but TS may have different compile destination. We expect all app code separated to .js + .d.ts. TypeScript can check js files against types.

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/impress/commit/e0e016e1417ee269704d99df48088bc1ebe38d88

**Pull request closed without merging**